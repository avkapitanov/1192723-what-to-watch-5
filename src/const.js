export const DEFAULT_RATING_FORM_VALUE = 3;
export const MAX_RATING_VALUE_FORM = 5;
export const HOVER_TIME_BEFORE_PLAYING = 2000;
export const FilmTab = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};
export const DEFAULT_ACTIVE_TAB = FilmTab.OVERVIEW;
export const DEFAULT_GENRE_FILTER_VALUE = `All genres`;
export const INITIAL_FILMS_COUNT = 8;
export const FILMS_COUNT_PER_STEP = 8;
export const MAX_GENRES_IN_FILTER = 9;
export const API_ENDPOINT = `https://5.react.pages.academy/wtw`;
export const REQUEST_TIMEOUT = 5000;
export const HttpResponseCode = {
  UNAUTHORIZED: 401
};
export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};
export const FilmRating = {
  BAD: {
    VALUE: 3,
    TEXT: `Bad`
  },
  NORMAL: {
    VALUE: 5,
    TEXT: `Normal`
  },
  GOOD: {
    VALUE: 8,
    TEXT: `Good`
  },
  VERY_GOOD: {
    VALUE: 10,
    TEXT: `Very good`
  },
  AWESOME: {
    TEXT: `Awesome`
  }
};

