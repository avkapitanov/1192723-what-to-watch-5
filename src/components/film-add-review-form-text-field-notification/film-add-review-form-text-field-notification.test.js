import React from "react";
import renderer from "react-test-renderer";
import FilmAddReviewFormTextFieldNotification from "./film-add-review-form-text-field-notification";
import {MAX_REVIEW_TEXT, MIN_REVIEW_TEXT} from "../../const";

describe(`FilmAddReviewFormTextFieldNotification component render correctly`, () => {
  it(`With min length error`, () => {
    const tree = renderer.create(
        <FilmAddReviewFormTextFieldNotification
          textLength={MIN_REVIEW_TEXT - 1}
          minLength={MIN_REVIEW_TEXT}
          maxLength={MAX_REVIEW_TEXT}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Without error`, () => {
    const tree = renderer.create(
        <FilmAddReviewFormTextFieldNotification
          textLength={MIN_REVIEW_TEXT}
          minLength={MIN_REVIEW_TEXT}
          maxLength={MAX_REVIEW_TEXT}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Without error in interval`, () => {
    const tree = renderer.create(
        <FilmAddReviewFormTextFieldNotification
          textLength={MAX_REVIEW_TEXT - MIN_REVIEW_TEXT}
          minLength={MIN_REVIEW_TEXT}
          maxLength={MAX_REVIEW_TEXT}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With max length error`, () => {
    const tree = renderer.create(
        <FilmAddReviewFormTextFieldNotification
          textLength={MAX_REVIEW_TEXT + 1}
          minLength={MIN_REVIEW_TEXT}
          maxLength={MAX_REVIEW_TEXT}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});


