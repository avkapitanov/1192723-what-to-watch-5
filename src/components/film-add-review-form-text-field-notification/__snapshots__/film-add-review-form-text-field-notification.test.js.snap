// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmAddReviewFormTextFieldNotification component render correctly With max length error 1`] = `
<div
  style={
    Object {
      "color": "red",
      "marginRight": "auto",
    }
  }
>
  Max symbols 400. Current: 401
</div>
`;

exports[`FilmAddReviewFormTextFieldNotification component render correctly With min length error 1`] = `
<div
  style={
    Object {
      "color": "red",
      "marginRight": "auto",
    }
  }
>
  Min symbols 50. Current: 49
</div>
`;

exports[`FilmAddReviewFormTextFieldNotification component render correctly Without error 1`] = `null`;

exports[`FilmAddReviewFormTextFieldNotification component render correctly Without error in interval 1`] = `null`;
